### ISSUE TITLE
Automatically add new entries to order?

### ISSUE THREAD

**Issue by chadcrowell** [AcclaroInc/craft-translations#160]
We create all new content on the site in english, and once a month, send those out for translation. We're now running Craft and I am testing out this translations plugin. Is there a way, or could you add a feature so that, whenever a new entry is created, it is automatically added to the current pending order?  

If not, is there a command I could execute in a plugin on a new entry save event to add it to an order?

Thanks.

**Comment by ifrimere** [comment]
@chadcrowell - thank you for submitting this feature request. I'm not sure about whether a Craft command (rule) could be set up to access the Translations plugin, find the 'pending order' and add the new Entry to it.

Our dev team will evaluate this feature request and if we can get a straight forward solution into an upcoming sprint, we will let you know. Besides adding the newly saved Entry to a 'pending Translations order' are there any other 'rules' you'd be interested in having - ex. what about 'saving changes to an existing Entry' or if no pending order exists, would you want the order automatically created.  Thanks.

fyi @sidedwards

**Comment by chadcrowell** [comment]
Thanks for the response. Honestly, yes, both automatically adding an edited entry to a pending order, and creating a new pending order if one doesn't exist would be very helpful. Beyond that:

1. Adding an edited global to a pending order
2. Ability to translate asset titles and custom fields by adding them to an order (and possibly auto adding them to pending order when uploaded, although editors would likely upload, and then edit the title/custom fields, so maybe when edited rather than uploaded)

This scenario would work well for us as we only ever have 1 pending order, but I realize it may not be ideal in all situations. Thank you.

**Comment by sidedwards** [comment]
Hi, @chadcrowell thanks for reaching out.

> Is there a way, or could you add a feature so that, whenever a new entry is created, it is automatically added to the current pending order?

This sounds like it could be a helpful feature; what are your thoughts on a setting whereby the user can target new Entries _within selected Sections_ to give more control over what content should be automatically added to an order?

> Adding an edited global to a pending order

I think creating draft orders based on editing/publishing actions could be interesting in the broader context of an "Automation Rules & Workflows" feature.

> Ability to translate asset titles and custom fields.

The plugin currently supports translating localized Asset fields when attached to an Entry.
<img width="597" alt="image" src="https://user-images.githubusercontent.com/11489608/118165011-03575280-b3e1-11eb-9efd-c849d91219ce.png">

Hopefully, this helps.

**Comment by chadcrowell** [comment]
> the user can target new Entries within selected Sections to give more control over what content should be automatically added to an order?

Yes, great idea. Some sections don't get translated. Makes sense.

> I think creating draft orders based on editing/publishing actions could be interesting in the broader context of an "Automation Rules & Workflows" feature.

Sounds like a good direction. I know Craft evaluates changes to fields on an entry to decide whether a draft should be saved, and/or if the "do you really want to leave the page" dialog is popped up. Wondering if you could leverage the same to detect if any fields changed (or even fields marked as translatable) on entry/global/asset save and add it to a pending order if there were changes.

> The plugin currently supports translating localized Asset fields when attached to an Entry.

How about assets not attached to an entry that have localized fields? We do have assets that show up in a site search, for instance (think documentation search results) that aren't attached to an entry.

Thanks very much, this is all sounding promising.

**Comment by sidedwards** [comment]
@chadcrowell thanks for the response.

We'll continue to scope out a feature for some of the automation features discussed; we'll aim to include it in an upcoming `v2.0` release of the plugin.

> How about assets not attached to an entry that have localized fields?

The plugin does not currently support this, but I think it's a great idea. If you could open a separate ticket with details on desired functionality, we can track our progress there.

Thank you!
